sui_core/src/authority/authority_store.rs

```diff
@@ -381,22 +352,22 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
     // Methods to mutate the store
 
     /// Insert a genesis object.
-    pub fn insert_genesis_object(&self, object: Object) -> SuiResult {
+    pub async fn insert_genesis_object(&self, object: Object) -> SuiResult {
         // We only side load objects with a genesis parent transaction.
         debug_assert!(object.previous_transaction == TransactionDigest::genesis());
         let object_ref = object.compute_object_reference();
+        self.lock_service.initialize_locks(vec![object_ref]).await?;
         self.insert_object_direct(object_ref, &object)
     }
 
-    /// Insert an object directly into the store, and also update relevant tables, including
-    /// initiating the transaction lock. This is used by the gateway to insert object directly.
+    /// Insert an object directly into the store, and also update relevant tables
+    /// NOTE: does not handle transaction lock.
+    /// This is used by the gateway to insert object directly.
     /// TODO: We need this today because we don't have another way to sync an account.
     pub fn insert_object_direct(&self, object_ref: ObjectRef, object: &Object) -> SuiResult {
         // Insert object
         self.objects.insert(&object_ref.0, object)?;
 
-        self.transaction_lock.get_or_insert(&object_ref, || None)?;
```

Comment:
This cannot be deleted. Both Gateway and genesis objects depend on this initializing locks for the objects.

Reply:
@lxfind thanks, I think this was an accident, I'll add an initialization call back in.
@gdanezis this is early, there are some tests failing probably because of this.  :)